apis/python/tests/test_sparse_nd_array.py

```diff
@@ -2007,3 +2007,38 @@ def test(path, tiledb_config):
             },
         )
         gc.collect()
+
+
+def test_sparse_nd_array_null(tmp_path):
+    uri = tmp_path.as_posix()
+
+    pydict = {
+        "soma_dim_0": pa.array([None, 1, 2, 3, 4, 5, 6, 7, 8, 9]),
+        "soma_data": pa.array(
+            [None, 0, None, 1, 2, None, None, 3, 4, 5], type=pa.float64()
+        ),
+    }
+    table = pa.Table.from_pydict(pydict)
+
+    soma.SparseNDArray.create(uri, type=pa.int64(), shape=(10,))
+
+    with soma.SparseNDArray.open(uri, "w") as A:
+        with raises_no_typeguard(soma.SOMAError):
+            # soma_joinid cannot be nullable
```

Comment:
As I understand, soma_data also can't be nullable - can we add a branch to test for both?

Reply:
https://github.com/single-cell-data/TileDB-SOMA/pull/3611/files#diff-e95b695dd56e1f3dc55f80eb74bc3f4fd55aefbe85ae1ee005e42ee6ae5732bfR2041-R2044

Attributes with null values are casted to fill values.